### PR TITLE
Migrate key management

### PR DESCRIPTION
**Pull request by Lagovas**
Migrated this section - https://docs.cossacklabs.com/pages/documentation-acra/#key-management

Updated information about newly added key types and purposes (symmetric keys, logging, searching). Changed mentions from `keys folder` to `keys storage` to abstract from a place where keys may be stored. Because now it is not the only filesystem.

### CODE REVIEW

**Comment by vixentael** [comment]
Shouldn't this be part of https://github.com/cossacklabs/product-docs/tree/master/acra/keys (or vice versa)

**Comment by Lagovas** [comment]
> Shouldn't this be part of https://github.com/cossacklabs/product-docs/tree/master/acra/keys (or vice versa)
Ouch, my bad. Missed this section. I remembered that @ilammy described keystore v2, acra-keys tool in our docs and thought that this section only about it without revising *(( Plus "Key management" mention and title in our new struct inside `acra-in-depth` pushed me to think that we don't have it and need to migrate

I will update my PR and merge existing docs in acra/keys that @ilammy has written and update with description about new key types that will be available in new Acra release (symmetric keys for AcraBlock, searchable encryption, audit logs)

I change this PR to draft until finish merging and then will change back when it will be ready for review

**Comment by Lagovas** [comment]
updated examples and descriptions with acra-ee keys and migrated full table with all locations and filenames for components

File: acra/keys/inventory.md
Comment (ilammy) [inline acra/keys/inventory.md:66]: If there are keys then there are keys.

```suggestion
- **AcraWriter** should have the storage public keys. They are necessary for encrypting data into AcraStructs in such a way that would only be readable for AcraServer/AcraTranslator.
```

File: acra/keys/inventory.md
Comment (ilammy) [inline acra/keys/inventory.md:58]:
```suggestion
    - Storage and zone asymmetric private keys for decrypting AcraStructs generated by AcraWriter.
```

Comment (ilammy) [inline acra/keys/inventory.md:58]: I think it would be nice to mention symmetric keys for AcraBlocks for AcraServer and AcraTranslator.
```
- Storage symmetric keys for encrypting and decrypting AcraBlocks in transparent mode.
```
Or something like this.

File: acra/keys/inventory.md
Comment (ilammy) [inline acra/keys/inventory.md:82]: I think it would be nice to wrap key file names into backticks rendering them `monospace` to signify that these are file names.

Another thing that I would do is making it obvious that `clientid` is not use verbatim but rather a placeholder. Given that there are examples with specific client IDs below, I think it would be okay to use `${clientID}` or something like this to denote substitution. Hopefully, this would be enough of a hint.

Ditto for symmetric keys table below.

File: acra/keys/inventory.md
Comment (storojs72) [inline acra/keys/inventory.md:21]:
```suggestion
AcraServer in Transparent proxy mode and AcraTranslator uses symmetric keys for hash calculations to implement searchable encryption. You can read more details in our [scientific paper](https://eprint.iacr.org/2019/806.pdf).
```

File: acra/keys/inventory.md
Comment (storojs72) [inline acra/keys/inventory.md:52]:
```suggestion
Correct operation of each Acra's component requires set of cryptographic keys (default keystore is `.acrakeys` filesystem directory)
```

File: acra/keys/inventory.md
Comment (storojs72) [inline acra/keys/inventory.md:21]:
```suggestion
AcraServer in Transparent proxy mode and AcraTranslator uses symmetric keys for hash calculations to implement searchable encryption. You can find more details in our [scientific paper](https://eprint.iacr.org/2019/806.pdf).
```

File: acra/keys/inventory.md
Comment (storojs72) [inline acra/keys/inventory.md:83]:
```suggestion
| Poison record with AcraStruct | `poison_key` | AcraServer<br/>AcraTranslator | `poison_key.pub` | AcraServer<br/>AcraTranslator
```

File: acra/keys/inventory.md
Comment (storojs72) [inline acra/keys/inventory.md:54]: IMO, I would use unified `needs to have` or `should have` for each service

File: acra/keys/inventory.md
Comment (storojs72) [inline acra/keys/inventory.md:56]: IMO, I would not refer to key names (since they are actually introduced below in correspondent tables), but just explain their high-level purpose, e.g.:
**AcraServer** should have:
1) Keys for accepting connections from clients via SecureSession;
2) Keys for decrypting AcraStructs generated by AcraWriter;
3) Keys for encrypting/decrypting AcraBlocks in transparent mode;
...

**Review by storojs72 — COMMENTED** [review]
Some stylistic suggestions

File: acra/keys/inventory.md
Comment (Lagovas) [reply acra/keys/inventory.md:54]: agree, unified. left "should" only for AcraWriter because it's correct only in case with app side encryption

File: acra/keys/inventory.md
Comment (Lagovas) [reply acra/keys/inventory.md:56]: IMO, the more verbose explanation is better. Because we have different type of keys (asymmetric/symmetric) for same purposes (AcraStruct/AcraBlock encryption). Same for cases with zones.

**Comment by Lagovas** [comment]
@ilammy, @storojs72 any other comments? please approve if nothing to add

**Review by storojs72 — APPROVED** [review]
LGTM

**Pull request merged**
